### ISSUE TITLE
Add groups to IAM user

### ISSUE THREAD

**Issue by edfenergy-stevebowerman** [dynatrace-oss/terraform-provider-dynatrace#461]
**Is your feature request related to a problem? Please describe.**
Feature request to enable a resource to add groups to a user.

We are currently using the `dynatrace_iam_user` to create users and associate their groups.

We are also using SCIM for SSO authentication and have 2x challenges:

1. User logs in before we run the terraform - causes the TF to fail (duplicate)
2. Pipeline taking 30+ mins to run with around 250 users.

As such we are looking for an alternative approach which:

1. We allow the SCIM process to create the DT user automatically on login
2. We change our TF to NOT create the user, but to 'add groups to existing users'

As such we'd use our config data file as the source of email addresses, but we'd need a way of adding a list of groups (which are terraformed vi `dynatrace_iam_group`) to these users.

**Describe the solution you'd like**
A new IAM resource like `dynatrace_iam_user_group` which would utilize this DT API: https://docs.dynatrace.com/docs/dynatrace-api/account-management-api/user-management-api/put-user-groups

Something like: 

```
resource "dynatrace_iam_user_group" "iam_user_group" {
    for_each = local.users

    email = each.value.email
    groups = local.groups[email]
}
```

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello @edfenergy-stevebowerman,

What you've described makes perfectly sense. With the next release the resource `dynatrace_iam_user` will no longer error out in case a user with that email already exists. Instead it will just update the group memberships.

The fact that your pipeline is taking 30+ mins to run is likely the result of a rate limiter we had to introduce recently, when `api.dynatrace.com` is being accessed. We've noticed an unusual amount of `504 Gateway Timeout` responses coming back from that endpoint and the only way to cope with that was to ensure that the provider doesn't send more than one request per second.
I've been told, that the issues with the gateway `should` be resolved meanwhile, but I didn't have time yet to try that out.
